dlib/test/global_optimization.cpp

```diff
@@ -232,6 +233,15 @@ namespace
         DLIB_TEST(std::abs(result.x - 1) < 1e-9);
         print_spinner();
 
+        constexpr auto close_enough = -16.0;
+        auto early_result = find_max_global([&early_evals](double x){ early_evals++; return -std::pow(x-2,2.0); }, -10, 1, max_function_calls(10), 0.0, std::vector<function_evaluation>{}, [](double y){ return (y >= close_enough);});
+        dlog << LINFO << "(x-2)^2, bound at 1: " <<  trans(early_result.x);
+        DLIB_TEST(early_result.x.size()==1);
+        DLIB_TEST(std::abs(early_result.y) <= std::abs(close_enough));
+        DLIB_TEST(std::abs(early_result.x - 1) <= 4);
+        DLIB_TEST(normal_evals >= early_evals);
```

Comment:
This needs to be `DLIB_TEST(normal_evals > early_evals);` or we don't know if the stopping condition did anything.